src/services/EdFi.DataManagementService.Api.Tests.E2E/Features/SchoolYearValidation.feature

```diff
@@ -0,0 +1,208 @@
+    When sending a POST request to "/ed-fi/courseOfferings" with body
+    """
+        {
+        "localCourseCode": "string",
+        "courseReference": {
+            "courseCode": "string",
+            "educationOrganizationId": 0
+        },
+        "schoolReference": {
+            "schoolId": 0
+        },
+        "sessionReference": {
+            "schoolId": 0,
+            "schoolYear": 0,
+            "sessionName": "string"
+        }
+        }
+    """
+    Then the response code is 400
+
+
+  #The cohort year scenario is interesting because it is testing a situation where a resource has a collection of SchoolYear references.
+  #The StudentEducationOrganizationAssociation has this collection, via CohortYears.
+  @Ignore
+  Scenario: Handling the array with two valid cohorts
+    Verify that the API correctly processes the request when both CohortYears are valid.
+    Given 2 valid CohortYears
+    When sending a POST request to "/ed-fi/studentEducationOrganizationAssociations" with body
+    """
+    {
+        "educationOrganizationReference": {
+            "educationOrganizationId": 0
+        },
+        "studentReference": {
+            "studentUniqueId": "string"
+        },
+        "cohortYears": [
+            {
+            "cohortYearTypeDescriptor": "string",
+            "termDescriptor": "string",
+            "schoolYearTypeReference": {
+                "schoolYear": 0
+            }
+            },
+            {
+            "cohortYearTypeDescriptor": "string",
+            "termDescriptor": "string",
+            "schoolYearTypeReference": {
+                "schoolYear": 0
+            }
+            }
+        ]
+    }
+    """
+    Then the response code is 200
+
+  @Ignore
+  Scenario: Handling the array with 2 cohorts (1st valid / 2nd invalid)
+    Ensure clients can not post when 1 of the CohortYears is invalid
+    Given 2 CohortYears invalid and valid
+    When sending a POST request to "/ed-fi/studentEducationOrganizationAssociations" with body
+    """
+    {
+        "educationOrganizationReference": {
+            "educationOrganizationId": 0
+        },
+        "studentReference": {
+            "studentUniqueId": "string"
+        },
+        "cohortYears": [
+            {
+            "cohortYearTypeDescriptor": "string",
+            "termDescriptor": "string",
+            "schoolYearTypeReference": {
+                "schoolYear": 0
+            }
+            },
+            {
+            "cohortYearTypeDescriptor": "string",
+            "termDescriptor": "string",
+            "schoolYearTypeReference": {
+                "schoolYear": 0
+            }
+            }
+        ]
+    }
+    """
+    Then the response code is 400
+
+  @Ignore
+  Scenario: Handling the array with 2 cohorts (1st invalid / 2nd valid)
+    Ensure clients can not post when 1 of the CohortYears is invalid
+    Given 2 CohortYears valid and invalid
```

Comment:
In the "Given" clause, should only be setting up required data. Since the background or request hook at the top will setup the valid school years, then ther is no need for line 180.